ios/StatusPanel/Data/DataSource.swift

```diff
@@ -37,55 +47,36 @@ struct DataItemFlags: OptionSet {
 
 protocol DataItemBase : AnyObject {
 
-    var icon: String? { get }
     var prefix: String { get }
     var flags: DataItemFlags { get }
     var subText: String? { get }
 
     func getText(checkFit: (String) -> Bool) -> String
 }
```

Comment:
OK. So this PR has made me think a lot about our model object here and where we're making various decisions. Right now, it seems that our rendering decisions are split between the source, and the renderer. And I think I was only making those things worse by the direction I was originally thinking of going.

So, I'd like to try a thought experiment: let's say that our 'renderer' knows best. That matches with the approach of this PR which has the 'renderer' is giving the data source a policy decision up-front.

Since the policy decision is made early, it requires us to re-fetch the data whenever our rendering choices change, preventing us from caching data items and making it harder to do live previews when changing options. That's not particularly problematic right now, but might be an issue in the future if we fetch from remote web services. It also pushes some of the policy decision complexity onto the data sources–they have to know how understand how the `.header` flag interacts with the `DisplayContext`.

These concerns encourage me to explore a completely explicit static property-driven model with no dynamic methods.

First up, and not even thinking about the icons for a moment, the `getText(checkFit:)` method would potentially cause us problems if we want to serve up truly static data. So, we could replace that with:

```swift
protocol DataItemBase : AnyObject {
    var prefix: String? { get }
    var text: String { get }
    var shortText: String { get }
    var footnote: String? { get }
    var flags: DataItemFlags? { get }
}

extension DataItemBase {
    func getText(checkFit: (String) -> Bool) -> String {
        checkFit(text) ? text : shortText
    }
}
```

This removes the need for any round-tripping to the data item to determine how to display something and makes our data item pure data. It does remove some flexibility, but if we're honest with ourselves, I doubt many data sources are going to support more than one sizing pass (as we've implemented in the calendar item). This change now means that a completely dumb web service can yield an item which complies with this (this example is just for you @tomsci):

```json
{
    "prefix": "💰",
    "text": "$44832.02 BTC/USD",
    "shortText": "$44832.02",
    "flags": []
}
```

So, pulling on this thread a little further, can we do something similar for the icons where we don't need to inject render state and can leave the decision solely to the renderer?:

```swift
protocol DataItemBase : AnyObject {
    var icon: Character? { get }
    var date: Date? { get }
    var text: String { get }
    var shortText: String { get }
    var footnote: String? { get }
    var flags: DataItemFlags { get }
}
```

The above change replaces `prefix` with `icon` and `date`. It's a bit of a cheat to make both changes at once as they're somewhat unrelated, but then the goal is similar, I think: the idea is to make it explicit how the data is going to be used. Exposing the date (should it be called `time` instead?) makes it very explicit that this is a temporal item. It allows us to sort temporal items by date, and interleave temporal events from multiple sources (perhaps I live in Hawaii 🌺 and have a remote source that reliably predicts my rainstorms). It also allows us to ensure that the date formatting for all temporal items (24/12h clock) is consistent across sources.

So far, this actually all feels pretty good to me. We can model everything we've got in the shipping app prior to this PR, and our remote Bitcoin data source looks almost identical:

```json
{
    "icon": "💰",
    "text": "$44832.02 BTC/USD",
    "shortText": "$44832.02",
    "flags": []
}
```

(Aside: If we make the `flags` an optional, we can even drop this from the JSON data.)

Now that I feel pretty good about myself and this line of thinking, I start to hit up hard against the problems this PR is trying to address: how do we model the two variants of this data item with and without icon support? My goal continues to be modelling using static data which doesn't need to change when our render-time properties change (icons on/off, header font changes). With the model described so far, we'd need two-or-so variations (with no way to switch between them):

- Icons enabled:
   ```json
   {
       "icon": "💰",
       "text": "$44832.02 BTC/USD",
       "shortText": "$44832.02",
       "flags": []
   }
   ```
- Icons disabled:
   ```json
   {
       "text": "$44832.02 BTC/USD",
       "shortText": "$44832.02",
       "flags": []
   }
   ```
   Or perhaps for our more prefix-oriented users:
   ```json
   {
       "prefix": "BTC",
       "text": "$44832.02",
       "shortText": "$44832.02",
       "flags": []
   }
   ```

How about combining these representations into a single data item?

```json
{
    "representations": {
        "graphical": { "icon": "💰", "long": "$44832.02 BTC/USD", "short": "$44832.02" },
        "text": { "prefix": "BTC", "long": "$44832.02", "short": "$44832.02" }
    },
    "flags": []
}
```

The naming and subtitles of the `representations` key clearly need some work, but this starts to feel much clearer to me as a data source author: I can explicitly create all of the manifestations meaning I don't have to write any conditional code; and I know it will work if I provide the requested ones. I would expect to make these explicit and strongly typed in Swift, so it would be impossible to construct a data item without the required attributes.

Thinking further along these lines, there are different ways we could tweak this. For example, consider what we might do if we actually acknowledge we have fonts that only support ascii (convoluted example warning):

```json
{
    "date": "2021-10-09 15:23:00+8",
    "representations": {
        "text": { "long": "Enjoy a glass of wine in a café", "short": "Drink" },
        "ascii": { "long": "Enjoy a glass of wine in a cafe", "short": "Drink" }
    },
    "flags": []
}
```

It's clear as I write this that there are going to be other ways to model this data (e.g., should the icon always be pulled up to the top level as, like the date, it's always relevant?), but there's enough here for me to feel like it's probably the direction I'd like to pursue: don't inject render state injected into the data sources, and force the sources to dumbly return the different representations we need for different display contexts.

What do you think?